pimd/pim_autorp.c

```diff
@@ -670,10 +676,19 @@ static void autorp_send_discovery(struct event *evt)
 			&(autorp->send_discovery_timer));
 }
 
+static bool pim_autorp_socket_enable(struct pim_autorp *autorp);
+static bool pim_autorp_socket_disable(struct pim_autorp *autorp);
+
 static void autorp_send_discovery_on(struct pim_autorp *autorp)
 {
 	int interval = 5;
 
+	/* Make sure the socket is open and ready */
+	if (!pim_autorp_socket_enable(autorp)) {
+		zlog_warn("%s: AutoRP failed to open socket", __func__);
```

Comment:
failing to open a socket should be an error I think.

Reply:
I checked the sources, half pf the places use `zlog_warn`(), the other half use `zlog_err()`!  gotta love consistency.